ctapipe/io/datawriter.py

```diff
@@ -357,6 +369,9 @@ def _setup_writer(self):
         if self.write_parameters is False:
             writer.exclude("/dl1/event/telescope/images/.*", "image_mask")
 
+        if self.write_muons is False:
```

Comment:
I think this is not the way do do this. If `write_muons` is False, we should not do anything. The column exclusions are meant to ignore specific columns of containers we do want to write.

Reply:
I did this primarily to suppress warnings, that the muon container is not writable as there is nothing stored in there when the muon-analysis is disabled through `write_muons` == False. Is there another way to achieve this?

Reply:
if the muon analysis is disabled, there should be no call to write(), I would assume?

Reply:
In my understanding, the same is done for the parameters container (alongside some other)

```python
# avoid some warnings about unwritable columns (which here are just
# sub-containers)
writer.exclude("dl1/event/subarray/trigger", "tel")
writer.exclude("dl1/monitoring/subarray/pointing", "tel")
writer.exclude("/dl1/event/telescope/images/.*", "parameters")
```
        
As I have implemented the storage for the muon_parameters in reminiscence to the "normal" parameters as sub-container of the  `DL1CameraContainer`, the same Warning for unwritable columns arises even when no muons are written. Afaik, this warning is not given when actual data is stored in this field, hence the if-statement. However, excluding it in any case seems to do no harm so I'll at least change this to be consistent to the above.